DistroLauncher/WSLInfo.cpp

```diff
@@ -38,62 +44,97 @@ namespace Helpers {
+    }
+
+    bool WslInfo::isX11UnixSocketMounted() const
+    {
+        // Commands to check if:
+        const std::array cmds = {
+          L"ls -l /tmp/.X11-unix | grep wslg",    // symlink points to wslg.
+          L"ls -l /tmp/.X11-unix/",               // symlink resolves correctly.
+          L"ss -lx | grep \"/tmp/.X11-unix/X0\"", // socket is listening.
+          L"test ! -z $DISPLAY",                  // DISPLAY is set inside the distro.
+        };
+        // I'm sure is better to read this way than with the algorithm.
+        // NOLINTNEXTLINE(readability-use-anyofallof)
+        for (const auto* const cmd : cmds) {
+            if (!WslLaunchSuccess(cmd)) {
+                wprintf(L"Command %s failed.\n", cmd);
+                return false;
+            }
         }
 
-        return distributionVersion;
+        return true;
     }
 
-    inline bool isWslgEnabled() {
+    inline bool WslInfo::isWslgEnabled() const
+    {
         return isX11UnixSocketMounted();
     }
 
-    bool WslGraphicsSupported() {
+    DWORD WslInfo::getDistroSubsystemVersion() const
+    {
+        return distroVersion;
+    }
+
+    bool WslInfo::supportsGraphics() const
+    {
         // Could WSL 3 or greater exist in the future?
-        return (Helpers::isWslgEnabled() &&
-                Helpers::WslGetDistroSubsystemVersion() > 1);
+        return isWslgEnabled() && (getDistroSubsystemVersion() > 1);
     }
-     
-/* =========================== INTERNALS ================================== */
-    namespace {
-        // Returns true if the WSL successfully launches a list
-        // of commands to ensure X11 socket is properly mounted.
-        // One indication that the distro has graphical support enabled.
-        bool isX11UnixSocketMounted() {
-            const std::array cmds = {
-                L"ls -l /tmp/.X11-unix",
-                L"ss -lx | grep \"/tmp/.X11-unix/X0\"",
-            };
-            // I'm sure is better to read this way than with the algorithm.
-            // NOLINTNEXTLINE(readability-use-anyofallof)
-            for (const auto *const cmd : cmds) {
-                if (!WslLaunchSuccess(cmd)) {
-                    wprintf(L"Command %s failed.\n", cmd);
-                    return false;
-                }
-            }
 
-            return true;
-        } 
-    } // namespace.
+    WSL_DISTRIBUTION_FLAGS WslInfo::getConfigurationFlags() const
+    {
+        return configFlags;
+    }
+
+    ULONG WslInfo::currentUserId() const
+    {
+        return currentUid;
+    }
+
+    WslInfo::~WslInfo()
+    {
+        //if (environtmenVars != nullptr && envVarsCount > 0) {
+        //    // discards the env variable string array otherwise they will leak.
+        //    for (int64_t i = envVarsCount - 1; i >= 0; --i) {
+        //        CoTaskMemFree(static_cast<LPVOID>(environtmenVars[i]));
+        //    }
+        //}
+    }
 } // namespace Helpers.
```

Comment:
I'm not happy with this class. That constructor should not even exist, and the attempt to cache those values acquired from WSL API in a short lived object doesn't make much sense. I guess this mission better fits in a wrapper for WSL API Loader. That would make somewhat easier to do some auto tests and let the cached values live for a longer time. I'll push some updates to this exact piece.